src/components/message-item.vue

```diff
@@ -141,28 +97,30 @@ watch(() => [filteredMessages, props.streamedMessageText, props.messages],
             :items="filteredMessages" key-field="id" v-slot="{ item, index, active }">
             <DynamicScrollerItem :item="item" :active="active" :data-index="index">
                 <div v-if="active" :class="messageClass(item.role)">
-                    <ToolTip :targetId="'message-label-' + index">
-                        Copy Text</ToolTip>
-                    <div class="label" @click="copyText(item)" :id="'message-label-' + index">
-                        <RefreshCcw v-if="item.role === 'user'" class="icon" :size="18"
+                    <div class="message-header">
+                        <RefreshCcw v-if="item.role === 'user'" class="icon" :id="'message-refresh-' + index" :size="18"
                             :class="{ 'loading': isLoading && loadingIcon === index }"
-                            @click="$emit('regenerate-response', item.content), startLoading(index)" />
-                        <Trash v-if="item.role === 'user'" class="delete-icon" :size="18"
+                            @click.stop="$emit('regenerate-response', item.content), startLoading(index)" />
+                        <ToolTip v-if="item.role === 'user'" :targetId="'message-refresh-' + index">Regenerate</ToolTip>
+                        <Trash v-if="item.role === 'user'" class="icon delete-icon" :id="'message-trash-' + index" :size="18"
                             :class="{ 'loading': isLoading && loadingIcon === index }"
```

Comment:
Can just remove this `:class` entirely for the Delete Icon I think, I noticed the trash can icon was also spinning with the regen arrows 😆

Reply:
You're right. Must have started to fix it a dozen times and every single time, forgot to do it 😅